packages/web-components/fast-foundation/src/avatar/avatar.ts

```diff
@@ -0,0 +1,89 @@
+import { attr, SyntheticViewTemplate } from "@microsoft/fast-element";
+import { FoundationElement, FoundationElementDefinition } from "../foundation-element";
+
+type AvatarShape = "circle" | "square";
+
+/**
+ * Avatar configuration options
+ * @public
+ */
+export type AvatarOptions = FoundationElementDefinition & {
+    media?: string | SyntheticViewTemplate;
+};
+
+/**
+ * An Avatar Custom HTML Element
+ *
+ * @public
+ */
+export class Avatar extends FoundationElement {
+    /**
+     * Indicates the Avatar should have a name.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: name
+     */
+    @attr public name: string;
+    protected nameChanged(): void {
+        if (!this.initials) {
+            const result = this.name?.match(/\b\w/g) || [];
+            this.initials = ((result.shift() || "") + (result.pop() || "")).toUpperCase();
+        }
+    }
+
+    /**
+     * Indicates the Avatar should have initials.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: initials
+     */
+    @attr public initials: string;
+
+    /**
+     * Indicates the Avatar should have a color fill.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: fill
+     */
+    @attr public fill: string;
```

Comment:
This is probably a bit of reworking, but I feel like these two colors should be based on the token infrastructure somehow. For instance, the badge color should probably adjust like an accent color for light or dark mode. This may be similar to how we handle a custom color for card, though I think it can be simpler than that if we create a new token that is a copy of the way accent fill works.

I see I didn't realize this on the presence badge PR, so I'll adjust there. Possibly something we want to clean up later, but I think it would be breaking.

Reply:
Could this still come in as an attribute and we update it in a future PR to adopt? While behavior is different, I see it as a feature which is additive to support accurate palettes...

For transparency, I'm curious if this implementation work needs to block vNext